server/src/main/java/io/crate/execution/jobs/RootTask.java

```diff
@@ -202,7 +201,11 @@ private CompletableFuture<Void> start(int taskIndex) {
             try {
                 logger.trace("Starting task job={} phase={} name={}", jobId, phaseId, task.name());
                 CompletableFuture<Void> started = task.start();
-                if (started != null) {
+                if (started == null || (started.isDone() && !started.isCompletedExceptionally())) {
```

Comment:
no need to change, just personally I find it easier to read:
```
if (started != null) {
    if (started.isCompletedExceptionally()) {
        return started;
    } else {
       return started.thenCompose(ignored -> start(taskIndex + 1));
    }
}
```

Reply:
That would still do the chaining for the `isDone()` (success) case. It would need:

```
                if (started != null) {
                    if (started.isDone()) {
                        continue;
                    } else if (started.isCompletedExceptionally()) {
                        return started;
                    } else {
                        return started.thenCompose(ignored -> start(taskIndex + 1));
                    }
                }
```


Which doesn't seem simpler to me